app/src/main/java/dadeindustries/game/gc/mechanics/turn/TurnProcessor.java

```diff
@@ -32,10 +35,56 @@ private void computeAiTurns(GlobalGameData globalGameData) {
 		}
 	}
 
-	public boolean processTurn(GlobalGameData globalGameData) {
+	private ArrayList<String> battleSummaries = new ArrayList<String>();
+
+	private Event processBattle(Sector s) {
+
+		int numberOfFactions = Faction.values().length;
+
+		Event result = new Event(Event.EventType.BATTLE, "Enemy ships encountered",
+				new Coordinates(s.getX(), s.getY()));
+
+		assert(s.getUnits().size() > 1);
```

Comment:
Asserts are something for tests, as such this is a good example of a test case, and something that the code should handle without the program exiting. 

Additionally, it is worth considering adding the @NotNull annotation to the processBattle signature in the parameters (though I leave this up to you, as annotations can messy if you add too many).

Reply:
Didn't know about that annotation, thanks.